File: backend/src/main/java/com/ead/backend/entity/Vehicle.java
package com.ead.backend.entity;

import jakarta.persistence.*;
import lombok.Data;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "vehicle")
@Data
@NoArgsConstructor
public class Vehicle {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    @Column(columnDefinition = "UUID")
    private UUID id;

    @Column(nullable = false)
    private String brand;

    @Column(nullable = false)
    private String model;

    @Column(nullable = false)
    private String year;

    @Column(nullable = false)
    private String color;

    @Column
    private LocalDateTime lastServiceDate;

    @Column(nullable = false, unique = true)
    private String licensePlate;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;
}

File: frontend/src/types/vehicle.ts
export interface Vehicle {
  id: string; // UUID as string
  brand: string;
  model: string;
  year: string;
  color: string;
  licensePlate: string;
  lastServiceDate?: string;
}

export interface VehicleCreateDto {
  brand: string;
  model: string;
  year: string;
  color: string;
  licensePlate: string;
  lastServiceDate?: string;
}

// No default export — only named exports for types
